### DISCUSSION TITLE
Pyodide in React: pyodide.globals - get all?

### DISCUSSION THREAD

**Discussion by ChristopherHButler** [pyodide/pyodide#1290] [Q&A]
In another discussion I mentioned I was interested in using pyodide for an online code editor I'm building. I finally got around to looking into this and managed to create a basic repl of sorts. You can see the result here: [https://contrived-dev.herokuapp.com/](https://contrived-dev.herokuapp.com/) by clicking just code (if you dont want to login with GitHub) and selecting vanilla python. (Sorry this is the dev env and python support is in active development so things may break).

Without rambling too much, I want to improve this functionality by letting users create a small app (like a codesandbox) in either vanilla javascript / typescript, react, vue, angular, etc. and have access to the functions, variables, etc. they create in python. My thought was to expose the pyodide.globals object in some way so users could get access to their python code. However, all the discussions I see around integrating pyodide with react seem to import react from js. I would like to access python from javascript, (specifically react as MY app is written with react). Is this possible? or do I HAVE to use pyodide.globals like `pyodide.globals.get("x");`. More to the point: I will not know what variables or functions a user creates so I cannot do something like `pyodide.globals.get("x");`. 

Is there anyone interested in discussing this with me to point me in the right direction? 

Thanks!

**Comment by ChristopherHButler** [comment]
I got a good response from rth on [stack overflow ](https://stackoverflow.com/questions/66403630/how-to-get-all-pyodide-globals-in-javascript/66440285#66440285) so this can be closed.

**Reply by rth** [reply]
That works. I don't think GH Discussions can be closed, it's kind of like a mailing list.
